### PR TITLE
Detect non-abstract methods with no body

### PR DESCRIPTION
**Pull request by paulbalandan**
Fixes https://github.com/phpstan/phpstan/issues/4244

### CODE REVIEW

File: src/Rules/Methods/AbstractMethodInNonAbstractClassRule.php [hunk 1]
```diff
@@ -20,24 +20,39 @@ public function getNodeType(): string
 		return Node\Stmt\ClassMethod::class;
 	}
 
+	/**
+	 * @param Node\Stmt\ClassMethod $node
```
Comment (ondrejmirtes) [inline src/Rules/Methods/AbstractMethodInNonAbstractClassRule.php]: This PHPDoc is not needed. Both PHPStan and PhpStorm know what's going into the method.

File: tests/PHPStan/Rules/Methods/AbstractMethodInNonAbstractClassRuleTest.php [hunk 2]
```diff
@@ -20,15 +20,22 @@ protected function getRule(): Rule
 
 	public function testRule(): void
 	{
-		$this->analyse([__DIR__ . '/data/abstract-method.php'], [
+		$this->analyse([
+			__DIR__ . '/data/abstract-method.php',
+			__DIR__ . '/data/bug-4244.php',
```
Comment (ondrejmirtes) [inline tests/PHPStan/Rules/Methods/AbstractMethodInNonAbstractClassRuleTest.php]: Please add a separate test method for testing a new file.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Rules/Methods/AbstractMethodInNonAbstractClassRule.php (hunk 1 above)
Comment (paulbalandan) [reply src/Rules/Methods/AbstractMethodInNonAbstractClassRule.php]: Done.

File: tests/PHPStan/Rules/Methods/AbstractMethodInNonAbstractClassRuleTest.php (hunk 2 above)
Comment (paulbalandan) [reply tests/PHPStan/Rules/Methods/AbstractMethodInNonAbstractClassRuleTest.php]: Done.

**Comment by ondrejmirtes** [comment]
Awesome, thank you!

**Pull request merged**